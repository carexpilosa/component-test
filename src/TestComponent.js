import React from 'react';

export default class TestComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      showHint: false
    };
  }

  render() {
    return (
      <div>
        <h3>TestComponent</h3>
        <button onClick={this.showTheHint.bind(this)}>button</button>
        {
          this.props.more ?
            <div>
                More: Ein Text und so weiter
            </div> :
            null
        }
        {
          this.state.hint ?
            <div>Hint: A Test Component :</div> :
            null
        }
      </div>
    );
  }

  showTheHint() {
    this.setState({
      showHint: true
    });
  }
}