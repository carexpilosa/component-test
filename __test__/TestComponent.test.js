import React from 'react';
import TestComponent from '../src/TestComponent';
import renderer from 'react-test-renderer';

test('Snapshot of component without  props', () => {
  let component = renderer.create(
    <TestComponent />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component = renderer.create(
    <TestComponent more={true}/>,
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.toTree().instance.showTheHint();
  console.log(component.toTree().instance.state.showHint);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
