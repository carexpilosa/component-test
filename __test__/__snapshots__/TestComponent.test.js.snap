// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of component without  props 1`] = `
<div>
  <h3>
    TestComponent
  </h3>
  <button
    onClick={[Function]}
  >
    show hint
  </button>
  <br />
  <button
    onClick={[Function]}
  >
    hide hint
  </button>
</div>
`;

exports[`Snapshot of component without  props 2`] = `
<div>
  <h3>
    TestComponent
  </h3>
  <button
    onClick={[Function]}
  >
    show hint
  </button>
  <br />
  <button
    onClick={[Function]}
  >
    hide hint
  </button>
  <div>
    More: Ein Text und so weiter
  </div>
</div>
`;

exports[`Snapshot of component without  props 3`] = `
<div>
  <h3>
    TestComponent
  </h3>
  <button
    onClick={[Function]}
  >
    show hint
  </button>
  <br />
  <button
    onClick={[Function]}
  >
    hide hint
  </button>
  <div>
    More: Ein Text und so weiter
  </div>
  <div>
    Hint: A Test Component
  </div>
</div>
`;

exports[`Snapshot of component without  props 4`] = `
<div>
  <h3>
    TestComponent
  </h3>
  <button
    onClick={[Function]}
  >
    show hint
  </button>
  <br />
  <button
    onClick={[Function]}
  >
    hide hint
  </button>
  <div>
    More: Ein Text und so weiter
  </div>
  <div>
    Hint: A Test Component
  </div>
</div>
`;
